applications/DEMApplication/tests/test_DEM_2D_contact_models.py

```diff
@@ -0,0 +1,84 @@
+import os
+import shutil
+import KratosMultiphysics as Kratos
+from KratosMultiphysics import Logger
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.DEMApplication as DEMApplication
+import KratosMultiphysics.DEMApplication.DEM_analysis_stage
+import auxiliary_functions_for_tests
+
+# To add a new contact model test, create a "MaterialsDEM_Contact_Model_Name.json" file that uses the target contact model to be tested,
+# and update the reference results in the GetReferenceResults function accordingly.
+
+class ContactModels2DTestSolution(KratosMultiphysics.DEMApplication.DEM_analysis_stage.DEMAnalysisStage, KratosUnittest.TestCase):
+    @classmethod
+    def GetMainPath(self):
+        return os.path.join(os.path.dirname(os.path.realpath(__file__)), "DEM2D_contact_models_tests_files")
+
+    def GetProblemNameWithPath(self):
+        return os.path.join(self.main_path, self.DEM_parameters["problem_name"].GetString())
+
+    def Finalize(self):
+        self.CompareResults()
+        self.procedures.RemoveFoldersWithResults(str(self.main_path), str(self.problem_name), "")
+        super().Finalize()
+
+    def CompareResults(self):
+        node = 16
+        tol = 1e-3
+        ref = self.GetReferenceResults()
+        f_ref = ref["force"]
+        t_ref = ref["torque"]
+        f = self.spheres_model_part.GetNode(node).GetSolutionStepValue(DEMApplication.CONTACT_FORCES)
+        t = self.spheres_model_part.GetNode(node).GetSolutionStepValue(DEMApplication.PARTICLE_MOMENT)
+        #print(self.DEM_parameters["solver_settings"]["material_import_settings"]["materials_filename"].GetString())
+        #print("Force:  [{:.15f}, {:.15f}, {:.15f}]".format(f[0],f[1],f[2]))
+        #print("Moment: [{:.15f}, {:.15f}, {:.15f}]".format(t[0],t[1],t[2]))
+        self.assertLess(abs((f[0]-f_ref[0])/f_ref[0]) if f_ref[0] != 0 else abs(f[0]-f_ref[0]), tol)
```

Comment:
In other test_case.py files, the function assertAlmostEqual() was usually used.
`self.assertAlmostEqual(x_force, x_force_ref, delta=tol)`
I know that both of them can achieve the same function, and I don't want to be picky about those minor details. But I think it could be better if we could use the same function, considering both the meaning and easy understanding. What do you think about it?

Reply:
I think it is better to check the relative difference between current and reference values (unless the reference is zero). That's why I used _assertLess_.
If I change it to _assertAlmostEqual_, the equivalent would be like:
_self.assertAlmostEqual(f[0], f_ref[0], delta = abs(f_ref[0]) * tol if f_ref[0] != 0 else tol)_
What do you think?

Reply:
But then it is not comparing the relative difference. Anyway, I modified it to a direct comparison.

Reply:
Thx. In the previous one, the same tolerance is used for both relative difference and absolute difference